File: Codeforces/Python/1462B-Last Year's Substring/solution.py
# https://codeforces.com/problemset/problem/1462/B

import sys

input = sys.stdin.readline

t = int(input())


def is_substring(s: str, target: str) -> bool:
    return (
        s[:4] == target or
        s[:3] + s[-1:] == target or
        s[:2] + s[-2:] == target or
        s[:1] + s[-3:] == target or
        s[-4:] == target
    )


for _ in range(t):
    n = int(input())
    s = input().rstrip()

    print("YES" if is_substring(s, "2020") else "NO")

File: Codeforces/C++/1462B-Last Year's Substring/solution.cpp
// https://codeforces.com/problemset/problem/1462/B

#include <iostream>
#include <string>

int main()
{
    int t;
    std::cin >> t;
    std::string target = "2020";

    while (t--)
    {
        int n;
        std::cin >> n;
        std::string str;
        std::cin >> str;

        bool found = false;

        for (int i = 0; i <= 4; ++i)
        {
            std::string prefix = str.substr(0, i);
            std::string suffix = str.substr(n - (4 - i), 4 - i);
            if (prefix + suffix == target)
            {
                found = true;
                break;
            }
        }

        std::cout << (found ? "YES" : "NO") << "\n";
    }

    return 0;
}
